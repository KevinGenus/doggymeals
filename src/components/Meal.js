import React, { Component } from "react";
import {
  Form,
  FormGroup,
  FormControl,
  FormLabel,
  Button
} from "react-bootstrap";

class Meal extends Component {
  constructor() {
    super();
    this.state = {
      dog: "",
      meal: ""
    };
  }
  render() {
    return (
      <div>
        <Form>
          <FormGroup>
            <FormLabel>Dog</FormLabel>
            <FormControl
              className="input-dog"
              onChange={event => this.setState({ dog: event.target.value })}
            />
          </FormGroup>
          <FormGroup>
            <FormLabel>Meal</FormLabel>
            <FormControl
              className="input-dog-meal"
              onChange={event => this.setState({ meal: event.target.value })}
            />
          </FormGroup>
        </Form>
        <Button
          className="btn-meal-remove"
          onClick={() => this.props.removeMeal(this.props.meal.id)}
        >
          Remove Meal
        </Button>
      </div>
    );
  }
}

export default Meal;
