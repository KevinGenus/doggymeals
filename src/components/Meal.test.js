import React from "react";
import { shallow } from "enzyme";
import Meal from "./Meal";
import toJson from "enzyme-to-json";

describe("Meal", () => {
  const mockRemoveMeal = jest.fn();
  const id = 1;
  const props = { meal: { id }, removeMeal: mockRemoveMeal };
  const meal = shallow(<Meal {...props} />);

  it("renders properly", () => {
    expect(toJson(meal)).toMatchSnapshot();
  });

  it("initializes a person and present in `state`", () => {
    expect(meal.state()).toEqual({ dog: "", meal: "" });
  });

  describe("when typing into the dog input", () => {
    const dogName = "Scooby Doo";
    beforeEach(() => {
      meal
        .find(".input-dog")
        .simulate("change", { target: { value: dogName } });
    });

    it("updates the doggy in `state`", () => {
      expect(meal.state().dog).toEqual(dogName);
    });
  });

  describe("when typing into the meal input", () => {
    const dogMeal = "Scooby Snack";
    beforeEach(() => {
      meal
        .find(".input-dog-meal")
        .simulate("change", { target: { value: dogMeal } });
    });

    it("updates the meal in `state`", () => {
      expect(meal.state().meal).toEqual(dogMeal);
    });
  });

  describe("when clicking the `Remove Meal` button", () => {
    beforeEach(() => {
      meal.find(".btn-meal-remove").simulate("click");
    });

    it("calls the removeMeal callback", () => {
      expect(mockRemoveMeal).toHaveBeenCalledWith(id);
    });
  });
});
