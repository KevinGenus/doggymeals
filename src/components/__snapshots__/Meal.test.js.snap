// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Meal renders properly 1`] = `
<div>
  <Form
    inline={false}
  >
    <FormGroup>
      <FormLabel
        column={false}
        srOnly={false}
      >
        Dog
      </FormLabel>
      <FormControl
        className="input-dog"
        onChange={[Function]}
      />
    </FormGroup>
    <FormGroup>
      <FormLabel
        column={false}
        srOnly={false}
      >
        Meal
      </FormLabel>
      <FormControl
        className="input-dog-meal"
        onChange={[Function]}
      />
    </FormGroup>
  </Form>
  <Button
    active={false}
    className="btn-meal-remove"
    disabled={false}
    onClick={[Function]}
    type="button"
    variant="primary"
  >
    Remove Meal
  </Button>
</div>
`;
